instrumentation/kubernetes-client-7.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/kubernetesclient/KubernetesClientInstrumentationModule.java

```diff
@@ -37,40 +43,97 @@ public KubernetesClientInstrumentationModule() {
   }
 
   public static class ApiClientInstrumentation implements TypeInstrumentation {
-    @Override
-    public ElementMatcher<ClassLoader> classLoaderOptimization() {
-      return hasClassesNamed("io.kubernetes.client.openapi.ApiClient");
-    }
-
     @Override
     public ElementMatcher<TypeDescription> typeMatcher() {
-      return extendsClass(named("io.kubernetes.client.openapi.ApiClient"));
+      return named("io.kubernetes.client.openapi.ApiClient");
     }
 
     @Override
     public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
-      return singletonMap(
-          ElementMatchers.isMethod()
-              .and(named("initHttpClient"))
-              .and(ElementMatchers.takesArguments(1))
-              .and(ElementMatchers.takesArgument(0, named("java.util.List"))),
-          KubernetesClientInstrumentationModule.class.getName() + "$KubernetesAdvice");
+      Map<ElementMatcher<MethodDescription>, String> transformers = new HashMap<>();
+      transformers.put(
+          isPublic()
+              .and(named("execute"))
+              .and(takesArguments(2))
+              .and(takesArgument(0, named("okhttp3.Call"))),
+          KubernetesClientInstrumentationModule.class.getName() + "$ExecuteAdvice");
+      transformers.put(
+          isPublic()
+              .and(named("executeAsync"))
+              .and(takesArguments(3))
+              .and(takesArgument(0, named("okhttp3.Call")))
+              .and(takesArgument(2, named("io.kubernetes.client.openapi.ApiCallback"))),
+          KubernetesClientInstrumentationModule.class.getName() + "$ExecuteAsyncAdvice");
+      return transformers;
+    }
+  }
+
+  public static class ExecuteAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnter(
+        @Advice.Argument(0) Call httpCall,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+      Context parentContext = Java8BytecodeBridge.currentContext();
+      if (!tracer().shouldStartSpan(parentContext)) {
+        return;
+      }
+
+      Request request = httpCall.request();
+      context = tracer().startSpan(parentContext, request, request.newBuilder());
```

Comment:
I don't follow what we are doing with the instance from `newBuilder`?

Reply:
Inject the context into HTTP headers - `Request` is immutable.

Reply:
Good catch - I should have been more cautious when copying code from the OkHttp instrumentation 😂

Reply:
Fixed - I've also modified tests to verify that propagation works as expected.